spec/server/listener_metrics_spec.rb

```diff
@@ -135,23 +147,33 @@ def start_puma(config_file, metrics_enabled, puma_port, metrics_port)
         envs_str = envs.map { |env, val| "#{env}=#{val}" }.join(' ')
 
         # Send logs to /dev/null to avoid cluttering the output
-        start_ok = system("#{envs_str} bundle exec bin/3scale_backend start -p #{puma_port} > /dev/null &")
+        start_ok = system("#{envs_str} bundle exec bin/3scale_backend " +
+                          "-s #{server} start -p #{listener_port} 2> /dev/null &")
         raise 'Failed to start Puma' unless start_ok
         sleep(2) # Give it some time to be ready
       end
 
-      def stop_puma(puma_port)
-        system("bundle exec bin/3scale_backend stop -p #{puma_port}")
+      def stop_listener(port, server)
+        if server == :puma
+          system("bundle exec bin/3scale_backend stop -p #{port}")
+        else # stop not implemented in Falcon
+          # Need to send 2 SIGTERMs because a bug in falcon v0.35.x
+          # https://github.com/socketry/falcon/issues/109
+          2.times do
+            system("ps axf | grep falcon | grep -v \"grep\" | tr -s ' ' | cut -d ' ' -f 2 | xargs kill")
```

Comment:
Uuuuuugh. Note that `ps` does not come standard with some container images, in particular not with UBI 8 minimal. What you do here can be done in a single command `pidof falcon` with better guarantees (`pidof` is included in the procps package, just like `ps`).

Reply:
Not sure why that provides better guarantees. `pidof falcon` does not work, we'd need `pidof ruby` which could clash with other ruby processes in the dev container.

Reply:
Fine, falcon does not set its own process title. Still that line is too broad. At least you should drop `a` (since it lists other users' processes).

Reply:
(also, perhaps you'd want `pgrep`?)

Reply:
The `a` can be dropped. Regarding the other 2 I see the same problem that I mentioned above.

Reply:
pgrep -f :+1:

Reply:
Used `2.times { system("pkill -u #{Process.euid} -f \"ruby .*falcon\"") }` in the end.